### DISCUSSION TITLE
Autostart Joiner

### DISCUSSION THREAD

**Discussion by juliankaroliny** [HomeSmartMesh/sdk-hsm-sensortag#11] [Q&A] [answered]
Hey,

I tested your samples/openthread_tag_sensors on the NRF52DK.
For me somehow the Autostart Joiner optiones do not work:

CONFIG_OPENTHREAD_JOINER=y
CONFIG_OPENTHREAD_JOINER_AUTOSTART=y
CONFIG_OPENTHREAD_JOINER_PSKD="ABCDEF"

On the Commisioner side I tried to add "ABCDEF" but nothing happens, also changed the Panid of the joiner.
Is anything additional needed on either sides?
(if i change to fixed network credentials, the tag_sensor joins without any problems and the rest is working pretty good)

br
Julian

**Comment by 0xlee** [comment]
Hey, 

maybe this might help: https://github.com/zephyrproject-rtos/zephyr/issues/30433#issuecomment-738851240

Best
Dongha

**Comment by wassfila** [comment, marked as answer]
@juliankaroliny 
I know that the OpenThread documentation is kind of made for those who already know it and is tricky to follow. I added some detailed examples here 
https://www.homesmartmesh.com/docs/networks/thread/#commissioning

Try to make it work with cli devices first, then you have a reference to compare with, worst case with sniffer which gives a great insight. Once working, rest is to figure out what command line param goes to which config item, you need to keep checking if you software compiled it correctly and make sure you delete the whole device because previous runs write on nv memory and could override new config params, that wasted me lots of time, otherwisewise make sure you know how to systematically debug all aspects of the thing, that the fun part and how someone learns, any part you cannot debug is where the bug will hide
here some hints :
https://www.homesmartmesh.com/docs/frameworks/zephyr/

**Comment by juliankaroliny** [comment]
Sorry for the way to late response, I had to set this problem on the side for the moment.

It was really the issue that not everthing is deleted with a normal flash.
I tested the "ot factoryreset" command and it worked perfectly after that.

However, I additionally had to add 
CONFIG_OPENTHREAD_MANUAL_START=n
so that the node joins after the commissioning 

Thanks again for both answers they really helped me

BR
Julian
